direct/core/src/test/java/cz/o2/proxima/direct/storage/InMemStorage.java

```diff
@@ -537,25 +545,27 @@ private void handleFlushDataBaseOnPosition(
           scheduler.scheduleAtFixedRate(
               onIdle, IDLE_FLUSH_TIME, IDLE_FLUSH_TIME, TimeUnit.MILLISECONDS);
       onIdleRef.set(onIdleFuture);
-      AtomicReference<StreamElement> lastConsumed = new AtomicReference<>();
+      final ConcurrentMap<Partition, StreamElement> lastConsumedPerPartition =
```

Comment:
This probably doesn't need ConcurrentMap, because there is lock held when accessing it.

Reply:
seems correct 👍